changefeedccl: add test to identify system table dependencies

asg0451 opened the pull request:
Add test to ensure that we're protecting the
system tables required for changefeed execution
with PTS.

Part of: https://github.com/cockroachdb/cockroach/issues/128806

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135732)
<!-- Reviewable:end -->

asg0451 committed:
changefeedccl: add test to identify system table dependencies

Add test to ensure that we're protecting the
system tables required for changefeed execution
with PTS.

Part of: #128806

Release note: None

rafiss reviewed (approved):
this lgtm! just had a very minor optional comment

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135732#-:-OEFE0v248zB6EU961FM:bp9ht30)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @andyyang890 and @asg0451)

___
*[`pkg/ccl/changefeedccl/changefeed_test.go` line 283 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135732#-OEFDZG19oFg8glcAyMa:-OEFDZG2EKIOQoI0cLcz:bs5j32d) ([raw file](https://github.com/cockroachdb/cockroach/blob/c3f2b7ff5c8485f3f6b72f795a35cec05255845a/pkg/ccl/changefeedccl/changefeed_test.go#L283)):*
> ```Go
> 	// NOTE: not all the required tables will necessarily show up in every run due to caching. However we should always see SOME tables.
> 	require.NotEmpty(t, tableIDsAccessed)
> 	allowedTables := make(map[catid.DescID]struct{}, len(systemTablesToProtect))
> ```

super nit: the `systemTablesToProtect` list is tiny, and this check is test only. so we could save a bit of boilerplate code by avoiding the conversion to a map, and instead use `slices.Contains` here.


<!-- Sent from Reviewable.io -->

asg0451 commented:
bors r=rafiss
